float32 electronics_x
float32 width



